### ISSUE TITLE
type 'String' is not a subtype of type 'bool'

### ISSUE THREAD

**Issue by biklas7** [flutterchina/dio#369]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

- Dio Version: `2.1.12`

`flutter doctor -v`

```
[✓] Flutter (Channel beta, v1.7.8+hotfix.2, on Linux, locale en_US.UTF-8)
    • Flutter version 1.7.8+hotfix.2 at /home/antonio/Documents/flutter
    • Framework revision 2e540931f7 (7 days ago), 2019-07-02 09:31:07 -0700
    • Engine revision b1cb0d9e9b
    • Dart version 2.4.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at /home/antonio/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: /home/antonio/Documents/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Android Studio (version 3.6)
    • Android Studio at /home/antonio/Documents/android-studio
    • Flutter plugin version 36.0.7
    • Dart plugin version 191.7830
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.36.0)
    • VS Code at /usr/share/code
    • Flutter extension version 3.2.0

[✓] Connected device (1 available)
    • Android SDK built for x86 • emulator-5554 • android-x86 • Android 9 (API 28) (emulator)

• No issues found!
```
### Issue Description and Steps

```
DioError (DioError [DioErrorType.DEFAULT]: type 'String' is not a subtype of type 'bool'#0      FormData.length.writeMapLength.<anonymous closure> (package:dio/src/form_data.dart:105:18)
#1      List.forEach (dart:core-patch/growable_array.dart:278:8)
#2      FormData.length.writeMapLength (package:dio/src/form_data.dart:103:27)
#3      FormData.length.<anonymous closure>.<anonymous closure> (package:dio/src/form_data.dart:131:27)
#4      List.forEach (dart:core-patch/growable_array.dart:278:8)
#5      FormData.length.<anonymous closure> (package:dio/src/form_data.dart:127:29)
#6      __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:367:8)
#7      FormData.length (package:dio/src/form_data.dart:121:10)
#8      Dio._transformData (package:dio/src/dio.dart:853:23)
<asynchronous suspension>
#9      Dio._makeRequest (package:dio/src/dio.dart:738:26)
<asynchronous suspension>
#10     Dio._request.<anonymous closure>.<anonymous closure> (package:dio/src/dio.dart:711:22)
#11     _rootRunUnary (dart:async/zone.dart:1132:38)
#12     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#13     _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
#14     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
#15     Future._propagateToListeners (dart:async/future_impl.dart:668:32)
#16     Future._complete (dart:async/future_impl.dart:473:7)
#17     _SyncCompleter.complete (dart:async/future_impl.dart:51:12)
#18     _AsyncAwaitCompleter.complete (dart:async-patch/async_patch.dart:28:18)
#19     _completeOnAsyncReturn (dart:async-patch/async_patch.dart:294:13)
#20     Dio._executeInterceptors (package:dio/src/dio.dart)
<asynchronous suspension>
#21     Dio._request.<anonymous closure> (package:dio/src/dio.dart:703:20)
#22     Dio._checkIfNeedEnqueue (package:dio/src/dio.dart:990:22)
#23     Dio._request (package:dio/src/dio.dart:702:9)
<asynchronous suspension>
#24     Dio.request (package:dio/src/dio.dart:614:12)
<asynchronous suspension>
#25     Dio.put (package:dio/src/dio.dart:169:12)
#26     UserServiceImpl.edit (package:Condor/data/api/user/user_service_impl.dart:43:53)
<asynchronous suspension>
#27     UserRepositoryImpl.editProfile (package:Condor/data/repositories/user/user_repository_impl.dart:71:43)
<asynchronous suspension>
#28     EditProfileBloc._mapEditProfileToState (package:Condor/ui/pages/edit_profile/bloc/edit_profile_bloc.dart:70:29)
<asynchronous suspension>
#29     EditProfileBloc.mapEventToState (package:Condor/ui/pages/edit_profile/bloc/edit_profile_bloc.dart:26:14)
<asynchronous suspension>
#30     Bloc._bindStateSubject.<anonymous closure> (package:bloc/src/bloc.dart:127:14)
#31     Stream.asyncExpand.onListen.<anonymous closure> (dart:async/stream.dart:513:30)
#32     _rootRunUnary (dart:async/zone.dart:1132:38)
#33     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#34     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#35     _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#36     _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#37     _StreamImplEvents.handleNext (dart:async/stream_impl.dart:707:11)
#38     _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:667:7)
#39     _rootRun (dart:async/zone.dart:1120:38)
#40     _CustomZone.run (dart:async/zone.dart:1021:19)
#41     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#42     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#43     _rootRun (dart:async/zone.dart:1124:13)
#44     _CustomZone.run (dart:async/zone.dart:1021:19)
#45     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
#46     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
#47     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#48     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
)
```

**Comment by biklas7** [comment]
![image](https://user-images.githubusercontent.com/4192633/60904308-bd705700-a26a-11e9-8be7-e25f9eb92abc.png)

**Comment by wendux** [comment]
Fixed in 2.1.13

**Comment by biklas7** [comment]
@wendux I can confirm that it was indeed fixed on `2.1.13`. I'm closing the issue. Thanks for the quick fix and awesome package :+1:
